docs/usage/configuration/config_documentation.md

```diff
@@ -436,6 +436,14 @@ Sub-options for each listener include:
 * `x_forwarded`: Only valid for an 'http' listener. Set to true to use the X-Forwarded-For header as the client IP. Useful when Synapse is
    behind a reverse-proxy.
 
+* `request_id_header`: The header extracted from each incoming request that is
+   used as the basis for the request ID. The request ID used in logs and tracing to
+   correlate and match up requests. When unset, Synapse will generate automatic
+   sequential request IDs. This option is useful when Synapse is behind a
+   reverse-proxy. For example, if you use Cloudflare in front of Synapse, you
+   can set this as `"cf-ray"` to match up requests even when the Cloudflare
+   layer times out before Synapse is done.
```

Comment:
This does help trace timeouts, but I think it is more applicable than that. It seems a bit odd to call out specifically here.

Reply:
Removed 👍